test/dogma/rules_test.exs

```diff
@@ -0,0 +1,67 @@
+defmodule Dogma.RulesTest do
+  use ExUnit.Case
+
+  alias Dogma.Rules
+  alias Dogma.Script
+  alias Dogma.Test.AssertingFormatter
+
+  @formater_for_testing Dogma.Formatter.Simple
+
+  test "returns empty list when given no scripts" do
+    result = Rules.test([], @formater_for_testing)
+    assert result == []
+  end
+
+  test "Config can be used to define the rules used" do
+    Application.put_env(:dogma, :rule_set, Dogma.RuleSets.FakeRuleSet)
+    single_script = %Script{}
+    expected_script_after_run = %Script{errors: [:always_fake_error]}
+
+    result = Rules.test([single_script], @formater_for_testing)
+
+    assert result == [expected_script_after_run]
+  end
+
+  test "Formatter.script is called by Rules.test" do
+    Application.put_env(:dogma, :rule_set, Dogma.RuleSets.FakeRuleSet)
```

Comment:
Is this safe to do? I don't really like changing global state like this.

Is this the right way to be testing this?